agents-api/agents_api/queries/executions/get_execution.py

```diff
@@ -45,8 +45,8 @@
 async def get_execution(
     *,
     execution_id: UUID,
-) -> tuple[str, dict]:
+) -> tuple[str, list]:
     return (
```

Comment:
you can add a third return value "fetchrow" and add rewrap_exception with asyncpg.exceptions.NoDataFoundError, this will return 404 in case the execution doesn't exist.

Make sure to refactor the return type of the function, and you can look at get_agent to see an example.